saber/boss_access/boss_access.py

```diff
@@ -67,70 +64,42 @@ def boss_pull_cutout(args):
     EXP_NAME = args.exp
     CHAN_NAME = args.chan
 
+    # Use intern to get d_type and i_type
+    intern_chans = array("bossdb://" + COLL_NAME + '/' + EXP_NAME + '/' + CHAN_NAME)
+
     # Create or get a channel to write to
     chan_setup = ChannelResource(
-        CHAN_NAME, COLL_NAME, EXP_NAME, type=args.itype, datatype=args.dtype
+        CHAN_NAME,
+        COLL_NAME,
+        EXP_NAME,
+        type=intern_chans._channel.type,
+        datatype=intern_chans.dtype
     )
     try:
         chan_actual = rmt.get_project(chan_setup)
     except HTTPError:
         chan_actual = rmt.create_project(chan_setup)
-    # get coordinate frame to determine padding bounds
-    cfr = CoordinateFrameResource(args.coord)
-    cfr_actual = rmt.get_project(cfr)
-    x_min_bound = cfr_actual.x_start
-    x_max_bound = cfr_actual.x_stop
-    y_min_bound = cfr_actual.y_start
-    y_max_bound = cfr_actual.y_stop
-    z_min_bound = cfr_actual.z_start
-    z_max_bound = cfr_actual.z_stop
 
     print("Data model setup.")
 
-    xmin = np.max([x_min_bound, args.xmin - args.padding])
-    xmax = np.min([x_max_bound, args.xmax + args.padding])
-    x_rng = [xmin, xmax]
-    ymin = np.max([y_min_bound, args.ymin - args.padding])
-    ymax = np.min([y_max_bound, args.ymax + args.padding])
-    y_rng = [ymin, ymax]
-    zmin = np.max([z_min_bound, args.zmin - args.padding])
-    zmax = np.min([z_max_bound, args.zmax + args.padding])
-    z_rng = [zmin, zmax]
     # Verify that the cutout uploaded correctly.
-    attempts = 0
-    while attempts < 3:
-        try:
-            cutout_data = rmt.get_cutout(chan_actual, args.res, x_rng, y_rng, z_rng)
-            break
-        except HTTPError as e:
-            if attempts < 3:
-                attempts += 1
-                print("Obtained HTTP error from server. Trial {}".format(attempts))
-            else:
-                print("Failed 3 times: {}".format(e))
     # Data will be in Z,Y,X format
+    cutout_data = rmt.get_cutout(resource=chan_actual,
+                                 resolution=args.res,
+                                 x_range=[args.xmin, args.xmax],
+                                 y_range=[args.ymin, args.ymax],
+                                 z_range=[args.zmin, args.zmax]
+                                 )
+
     # Change to X,Y,Z for pipeline
     cutout_data = np.transpose(cutout_data, (2, 1, 0))
```

Comment:
do we want to make this a arg for the tool? data_order="xyz" or "zyx" with "xyz" as default? This has come up several time as a bug and we have to write transpose steps into other tools!

Reply:
this would only be for boss_pull_cutout right? when a user pushes data back to bossdb should we assume that their data is already in xyz order?

Reply:
data in bossdb are stored in ZYX order